streams/prngs/testu01-prngs/testu01_interface.h

```diff
@@ -0,0 +1,101 @@
+//
+// Created by mhajas on 11/14/17.
+//
+#pragma once
+
+#include <streams/prngs/prng_interface.h>
+#include <utility>
+
+extern "C" {
+#include <streams/prngs/testu01-prngs/includes/unif01.h>
+};
+
+namespace prng {
+
+    template<typename GENERATOR, typename DELETER, std::uint8_t OUTPUT_SIZE>
+    class testu01_interface : public prng_interface {
+    protected:
+        std::function<std::unique_ptr<GENERATOR, DELETER>(const value_type *)> _generator_creator;
+        std::unique_ptr<stream> _seeder;
+        vec_cview _seed;
+        std::unique_ptr<GENERATOR, DELETER> _generator;
+        bool _reseed_for_each_test_vector;
+
+    public:
+        explicit testu01_interface(const std::function<std::unique_ptr<GENERATOR, DELETER>(const value_type *)> &creator,
+                                   std::unique_ptr<stream> &seeder, bool reseed_for_each_test_vector)
+                : _generator_creator(std::move(creator))
+                , _seeder(std::move(seeder))
+                , _seed(_seeder->next())
+                , _generator(_generator_creator(_seed.data()))
+                , _reseed_for_each_test_vector(reseed_for_each_test_vector) {}
+
+        void generate_bits(std::uint8_t *data, size_t number_of_bytes) override {
+            int current_index = 0;
+
+            while (number_of_bytes > 0) {
+
+                uint64_t generated_data = _generator->GetBits(_generator->param,
+                                                              _generator->state); // Get new Bytes from generator
+
+                for (auto i = 0; i < std::min<size_t>(number_of_bytes, OUTPUT_SIZE); i++) {
+                    data[current_index++] = static_cast<std::uint8_t>(generated_data & 0xFF);
+                    generated_data >>= 8;
+                }
+
+                number_of_bytes -= std::min<size_t>(number_of_bytes, OUTPUT_SIZE);
+            }
+
+            if (_reseed_for_each_test_vector) {
+                _seed = _seeder->next();
+                _generator = _generator_creator(_seed.data());
+            }
+        }
+
+
+        /**
+         *
+         * In most cases we need to avoid seeds which are greater or equal to modulo, then it is quite hard to detect
+         * when this is violated and when not, therefore I decided not to generate seed for full 8 Bytes (long), but to
+         * use only seeds which are guaranteed not fulfill requirements, even though we loose some possible seeds.
+         *
+         *
+         * @tparam T
+         * @param number to investigate
+         * @return returns how many bytes from beginning are not zero
+         */
+        template<typename T>
+        static size_t get_viable_number_of_bytes(T number) {
```

Comment:
https://www.go4expert.com/articles/builtin-gcc-functions-builtinclz-t29238/
https://github.com/crocs-muni/eacirc/blob/master/eacirc/circuit/connectors.h

Reply:
This was the place, I wanted to refer to: https://github.com/crocs-muni/eacirc-core/blob/master/builtins.h